go/checks.rst

```diff
@@ -0,0 +1,302 @@
+| following values:                                                                                |
+| - ``"WARNING"`` : warning message emitted at compile-time                                        |
+| - ``"ERROR"``   : fails compilation in addition to emitting an error message                     |
++----------------------------+---------------------------------------------------------------------+
+| ``"apply_to"``             | :type:`dictionary, string to string`                                |
++----------------------------+---------------------------------------------------------------------+
+| Specifies files that this check will exclusively apply to.                                       |
+| Its keys are regular expression strings matching Go files, and its values are strings containing |
+| a description of the entry.                                                                      |
++----------------------------+---------------------------------------------------------------------+
+| ``"whitelist"``            | :type:`dictionary`                                                  |
++----------------------------+---------------------------------------------------------------------+
+| Specifies files that are exempt from this check.                                                 |
+| Its keys and values are strings that have the same semantics as those in `apply_to`.             |
+| Keys in whitelist override keys in apply_to. If a .go file matches both an `apply_to` and        |
+| `whitelist` key, the check will not apply to that file.                                          |
++----------------------------+---------------------------------------------------------------------+
+
+Example
+^^^^^^^
+
+The following configuration file configures the checks named ``mycheck``,
+``unsafedom``, and ``loopclosure``.
+
+.. code:: json
+
+    {
+      "mycheck": {
+        "severity": "ERROR",
+        "whitelist": {
+          "src/foo.go": "manually verified that behavior is working-as-intended"
+          "src/bar.go": "see issue #1337"
+        }
+      },
+      "unsafedom": {
+        "severity": "WARNING",
+        "apply_to": {
+          "src/js/*": ""
+        },
+        "whitelist": {
+          "src/(third_party|vendor)/*": "enforce DOM safety requirements only on first-party code"
+        }
+      },
+      "loopclosure": {
+        "description": "fail builds without exception since we know this check is 100% accurate",
+        "severity": "ERROR"
+      }
+    }
+
+This label referencing this configuration file must be provided as the
+``config`` attribute value of the ``go_checker`` rule.
+
+.. code:: bzl
+
+    go_checker(
+        name = "my_checker",
+        deps = [
+            ":mycheck",
+            ":unsafedom",
+            "@javascript_checks//:loopclosure",
+        ],
+        config = ":config.json"
```

Comment:
Nit: Don't include the leading `:` in the filename. It doesn't make any difference to Bazel, but it helps distinguish rules from static files.

Reply:
Done.